src/components/button/index.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { FC, ReactNode } from 'react';
+import {
+  ButtonProps,
+  Button as CUButton,
+  StyleProps,
+  useTheme,
+  useToken,
+} from '@chakra-ui/react';
+
+interface Props extends ButtonProps {
+  children: ReactNode;
+  size?: 'md' | 'lg' | 'xl';
+  variant?: 'primary';
+}
+interface ButtonStates {
+  normal: StyleProps;
+  hover: StyleProps;
+  active: StyleProps;
+  disabled: StyleProps;
+}
+
+const Button: FC<Props> = ({
+  children,
+  variant = 'primary',
+  size = 'lg',
+  ...rest
+}) => {
+  const btnShadowColor = useToken('colors', 'custom.btnShadow');
+
+  const { name } = useTheme();
+
+  let buttonStates: ButtonStates;
+
+  const fontSize = {
+    md: {
+      fontSize: 'sm',
+    },
+    lg: {
+      fontSize: 'md',
+    },
+    xl: {
+      fontSize: 'md',
+    },
+  };
+
+  const spacing = {
+    md: {
+      py: 'sm',
+      px: 'md',
+    },
+    lg: {
+      py: 'md',
+      px: 'lg',
+    },
+    xl: {
+      py: 'lg',
+      px: 'xl',
+    },
+  };
+
+  const commonNormalState: StyleProps = {
+    borderRadius: 'sm',
+    boxShadow: `0px 2px 0px ${btnShadowColor}`,
+    fontWeight: 'bold',
+    ...spacing[size],
+    ...fontSize[size],
+  };
+
+  const disabledState: StyleProps = {
+    bg: 'gray.100',
+    boxShadow: 'none',
+    color: 'gray.400',
+    pointerEvents: 'none',
+  };
+
+  if (name === 'AS24') {
```

Comment:
I might misunderstand something but isn't the idea to make use of the theme provided by chakra so the states can be defined once and the differences are provided by the theme?

Reply:
You're completely right, that's the idea. In the ideal case, the theme provider should control the brand color, depends on the theme name and we should not do the theme name check here. In our current situation, the styles for AS24/MS24 is little bit different for background & font color. Let's have a look:

```ts
const as24NormalState = {
  bg: 'brand.100',
  color: 'gray.900',
}

const ms24NormalState = {
  bg: 'brand.500',
  color: 'black',
}
```

The `brand` color token will be updated as soon as we switch the theme, but the value `100` should stay the same, between themes. The color for text is also different between themes as well. That's a reason why I have do theme name check for applying button styles. 

Do you have suggestion for me to improve the logic here?

Reply:
Maybe here is where we can make use of the `semantic tokens` -> https://chakra-ui.com/docs/styled-system/features/semantic-tokens